### PR TITLE
Adding a general cloud provider file for kubelet reset and fork bomb on all clouds

### PR DESCRIPTION
**Pull request by paigerube14**
This PR will allow a user to specify there correct cloud type for node scenarios to allow them to run the stop start kubelet scenario as well as the fork bomb scenario.

This is for https://github.com/openshift-scale/kraken/issues/46

### CODE REVIEW

**Review by chaitanyaenr — APPROVED** [review]
LGTM

File: run_kraken.py [hunk 1]
```diff
@@ -22,6 +23,8 @@ def get_node_scenario_object(node_scenario):
         return aws_node_scenarios()
     elif node_scenario['cloud_type'] == 'gcp':
         return gcp_node_scenarios()
+    else:
```
Comment (mffiedler) [inline run_kraken.py:26]: Nitpick: I think defaulting to 'generic' if cloud_type is not specified is fine, but let's not allow arbitrary values.   Something like:
if node_scenario['cloud_type'] == 'generic' or not 'cloud_type' in node_scenario:
   generic
else:
  bad cloud type

Comment (paigerube14) [reply run_kraken.py:28]: Think that is much better! Added some documentation to show this as well

File: run_kraken.py [hunk 2]
```diff
@@ -12,16 +12,24 @@
...
 
 # Get the node scenarios object of specfied cloud type
 def get_node_scenario_object(node_scenario):
+    if "cloud_type" not in node_scenario.keys() or node_scenario['cloud_type'] == "generic":
+        return general_node_scenarios()
     if node_scenario['cloud_type'] == 'aws':
         return aws_node_scenarios()
     elif node_scenario['cloud_type'] == 'gcp':
         return gcp_node_scenarios()
+    else:
```
Comment (mffiedler) [inline run_kraken.py:28]: Still using arbitrary values as 'generic'.  Missed a file on the last commit?

Comment (paigerube14) [reply run_kraken.py:28]: Sorry about that, somehow missed the main file in my last commit! Should be there now

**Comment by mffiedler** [comment]
I don't think we can use generic for stop_start_kubelet_scenario.   Starting the kubelet uses the reboot scenario which requires a cloud provider:

```
    def stop_start_kubelet_scenario(self, instance_kill_count, node, timeout):
        logging.info("Starting stop_start_kubelet_scenario injection")
        self.stop_kubelet_scenario(instance_kill_count, node, timeout)
        self.node_reboot_scenario(instance_kill_count, node, timeout)
        logging.info("stop_start_kubelet_scenario has been successfully injected!")
```

**Comment by paigerube14** [comment]
> I don't think we can use generic for stop_start_kubelet_scenario. Starting the kubelet uses the reboot scenario which requires a cloud provider:
> 
> ```
>     def stop_start_kubelet_scenario(self, instance_kill_count, node, timeout):
>         logging.info("Starting stop_start_kubelet_scenario injection")
>         self.stop_kubelet_scenario(instance_kill_count, node, timeout)
>         self.node_reboot_scenario(instance_kill_count, node, timeout)
>         logging.info("stop_start_kubelet_scenario has been successfully injected!")
> ```

Ah yes. It's interesting because the doc listed stop kubelet in needing a specfici cloud but then also listed as generic. Removed it from the generic area. Do you think there is more that we should do to stop people from using certain node actions for cloud types that we have not set up or that are "generic" for now?

File: docs/node_scenarios.md [hunk 3]
```diff
@@ -28,9 +28,9 @@ A google service account is required to give proper authentication to GCP for no
  
 After creating the service account you'll need to enable the account using the following: ```export GOOGLE_APPLICATION_CREDENTIALS="<serviceaccount.json>"```
 
-**NOTE**: The `stop_start_kubelet_scenario` and `node_crash_scenario` scenarios are supported as they are independent of the cloud platform.
+**NOTE**: The `node_crash_scenario` scenario is supported independent of the cloud platform.
```
Comment (mffiedler) [inline docs/node_scenarios.md:31]: Do you think we need to flag the scenario as not-applicable to the generic cloud_provider in the code.  i.e. treat it like reboot, etc?   Sorry to be so nitpicky about this

Comment (paigerube14) [reply docs/node_scenarios.md:31]: No worries! I was thinking abut that as well just to maybe inform the user of their incorrect scenario set up.

Comment (paigerube14) [reply docs/node_scenarios.md]: Added a lot of if statements here to log what is happening. Is this over kill?

**Comment by paigerube14** [comment]
I noticed that the tox spacing was incorrect for the merge in GCP. Updated in this pr. Is this ok or should I undo it?

File: docs/node_scenarios.md (hunk 3 above)
Comment (paigerube14) [reply docs/node_scenarios.md]: Found that the doc for generic scenarios should've said "stop_kubelet_scenario" instead of stop/start. Updated this

**Pull request merged**